packages/at_commons/lib/src/verb/update_verb_builder.dart

```diff
@@ -274,4 +216,65 @@ class UpdateVerbBuilder implements VerbBuilder {
     }
     return false;
   }
+
+  void _validateKey() {
+    if (isLocal == true && (isPublic == true || sharedWith.isNotNull)) {
+      throw InvalidAtKeyException(
+          'When isLocal is set to true, cannot set isPublic and sharedWith');
+    }
+    if (isPublic == true && sharedWith.isNotNull) {
+      throw InvalidAtKeyException(
+          'When isPublic is set to true, sharedWith cannot be populated');
+    }
+    if (atKey.isNull) {
+      throw InvalidAtKeyException('Key cannot be null or empty');
+    }
+  }
+
+  String buildKey() {
```

Comment:
This buildKey looks like a duplicate of the buildKey in llookup_verb_builder.dart

Please extract duplicate code into a single place.

Reply:
Added abstract_verb_builder.dart to move the redundant code to a common place.